File: Python_Ruby/object_oriented/tableformatter.rb
# tableformatter.rb
#
# Will MacLean
# CSPP 51060

class TableFormatter
	attr_reader :headers, :rowdata
	attr_writer :headers, :rowdata
	def initialize(headers, rowdata)
		@headers = headers
		@rowdata = rowdata
	end
	def headings(headers)
		newStr = ''
		for heading in @headers
			newStr = newStr + heading + '\t'
		end
		return newStr + '\n'
	end
	def row(rowdata)
		newStr = ''
		for row in @rowdata
			for field in row
				newStr = newStr + field.to_s + '\t'
			end
			newStr = newStr + '\n'
		end
		return newStr
	end
	def buildtable
		newStr = headings(@headers)
		newStr = newStr + row(@rowdata)
		writetable(newStr, 'table.txt')
		return newStr
	end
	def writetable(aTable, filename)
		fileOut = File.open(filename, 'w')
		fileOut.puts(aTable)
		fileOut.close
	end
end

class HtmlTableFormatter <TableFormatter
	def headings(headers)
		newStr = '<html><table><tr>'
		for heading in @headers
			newStr = newStr + '<td><b>' + heading + '</b></td>'
		end
		return newStr + '</tr>'
	end
	def row(rowdata)
		newStr = ''
		for row in @rowdata
			newStr = newStr + '<tr>'
			for field in row
				newStr = newStr + '<td>' + field.to_s + '</td>'
			end
			newStr = newStr + '</tr>'
		end
		return newStr
	end
	def endtable
		return '</table></html>'
	end
	def buildtable
		newStr = headings(@headers)
		newStr = newStr + row(@rowdata) + endtable
		writetable(newStr, 'table.html')
		return newStr
	end
end

class CsvTableFormatter <TableFormatter
        def headings(headers)
                newStr = ''
                for heading in @headers
                        newStr = newStr + heading + ','
                end
                return newStr.chomp(',') + '\n'
        end
        def row(rowdata)
                newStr = ''
                for row in @rowdata
                        for field in row
                                newStr = newStr + field.to_s + ','                        end
                        newStr = newStr.chomp(',') + '\n'
                end
                return newStr
        end
        def buildtable
                newStr = headings(@headers)
                newStr = newStr + row(@rowdata)
                writetable(newStr, 'table.csv')
                return newStr
        end
end

# test stuff
#heads = [ "name", "shares", "price", "change" ]
#rows = [ ["MSFT", 10, 26.00, -1.20],  ["ORCL", 20, 32.00, -2.20],  ["DDT", 30, 2.00, -0.20] ]

#test = CsvTableFormatter.new(heads, rows)
#print test.headings(heads)
#print test.row(rows)
#print test.buildtable



# this is just to move the prompt to the next line when done
printf "\n"

File: Python_Ruby/object_oriented/tableformatter.py
# tableformatter.py
#
# Will MacLean
# CSPP 51060

class TableFormatter(object):
	# the base class prints text tables
	def __init__(self, headers, rowdata):	# constructor
		self.headers = headers
		self.rowdata = rowdata
	def headings(self, headers):		# print heads
		newStr = ''
		print '\n'				# skip line
		for heading in self.headers:		# walk thru heads
			newStr = newStr + heading + '\t'
		newStr = newStr + '\n'
		for i in range(0, len(headers)):	# print lines
			newStr = newStr + '----- \t'
		newStr = newStr + '\n'
		return newStr
	def row(self, rowdata):			# print rows
		import testmod
		newStr = ''
		for row in self.rowdata:		# grab a row
			for field in row:		# grab 	 field
				temp = str(field)
				newStr = newStr + temp + '\t'
			newStr = newStr + '\n'
		return newStr
	def endtable(self):			# not needed in text
		pass
	def buildtable(self, headers, rowdata):
		newStr = self.headings(self.headers)
		newStr = newStr + self.row(self.rowdata)
		self.writetable(newStr, 'table.txt')
		return newStr
	def writetable(self, table, filename):
		fileOut = open(filename, "w")
		fileOut.write(table)
		fileOut.close()

class HtmlTableFormatter(TableFormatter):
	def headings(self, headers):
		newStr = '<html><table><tr>'
		for heading in self.headers:
			newStr = newStr + '<td><b>' + heading + '</b></td>'
		newStr = newStr + '</tr>'
		return newStr
	def row(self, rowdata):
		newStr = ''
		for row in self.rowdata:
			newStr = newStr + '<tr>'
			for field in row:
				newStr = newStr + '<td>' + str(field) + '</td>'
			newStr = newStr + '</tr>'
		return newStr
	def endtable(self):
		return '</table></html>'
	def buildtable(self, headers, rowdata):
		newStr = self.headings(self.headers)
		newStr = newStr + self.row(self.rowdata)
		newStr = newStr + self.endtable()
		self.writetable(newStr, 'table.html')
		return newStr

class CsvTableFormatter(TableFormatter):
	def headings(self, headers):
		newStr = ''
		for heading in self.headers:
			newStr = newStr + heading + ','
		return newStr.rstrip(',') + '\n'
	def row(self, rowdata):
		newStr = ''
		for row in self.rowdata:
			for field in row:
				newStr = newStr + str(field) + ','
			newStr = newStr.rstrip(',') + '\n'
		return newStr
	def buildtable(self, headers, rowdata):
		newStr = self.headings(self.headers)
		newStr = newStr + self.row(self.rowdata)
		self.writetable(newStr, 'table.csv')
		return newStr

heads = [ "name", "shares", "price", "change" ]
rows = [ ["MSFT", 10, 26.00, -1.20],  ["ORCL", 20, 32.00, -2.20],  ["DDT", 30, 2.00, -0.20] ]

test = CsvTableFormatter( heads, rows )
#print test.headings( heads )
#print test.row( rows )
table = test.buildtable(heads, rows)
print table


